src/compiler/checker.ts

```diff
@@ -43194,6 +43195,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isValidUnusedLocalDeclaration(declaration: Declaration): boolean {
+        if (isTypeAliasDeclaration(declaration)) {

```

Comment:
At some level, I wonder if we should just be checking `isIdentifierThatStartsWithUnderscore` for all declarations, rather than special casing a few kinds. What declarations are we actually missing here?

Reply:
I think this is open to discussion. The fix shouldn't require significant work... What do you think, @RyanCavanaugh?

Reply:
I guess it's sort of nonsensical to do this for enum members and properties and so on; after all, their names are important.

Reply:
@jakebailey What are your thoughts on the following list? Should we add or exclude any additional declarations?

1. Interface Declaration
2. Type Alias Declaration
3. Enum Declaration
4. Class Declaration

Reply:
I would say any declaration that can be a child of a `Block` or `SourceFile` or `Module`, etc, if that makes sense. So basically _not_ enum members or properties. Not sure we have a list of that variety, though...